app/src/main/java/com/example/calculator/Divide.java

```diff
@@ -8,7 +10,7 @@ public Divide() {
     public NumberToken calculate(Token num1, Token num2) {
         double num2safe = num2.getNumber();
         if(Double.compare(num2safe,0) == 0) {
-            num2safe = 1;
+            throw new ArithmeticException();
         }
```

Comment:
You don't need this by yourselves.
When you divide by zero, JVM will throw the exception.

Reply:
https://stackoverflow.com/questions/14137989/java-division-by-zero-doesnt-throw-an-arithmeticexception-why